### PR TITLE
Upgrade to TypeScript 2.6.1, turn on strict, and bump minor version

### PR DESCRIPTION
**Pull request by bryanforbes**
**Type:** enhancement

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Relates to dojo/meta#210

### CODE REVIEW

File: src/accordionpane/example/index.ts [hunk 1]
```diff
@@ -15,8 +15,9 @@ export class App extends WidgetBase<WidgetProperties> {
 	private _openKeys = new Set<string>();
 	private _theme: {};
 
-	themeChange(event: TypedTargetEvent<HTMLInputElement>) {
```
Comment (pottedmeat) [inline src/accordionpane/example/index.ts]: Why was this changed?

File: src/accordionpane/example/index.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ export class App extends WidgetBase<WidgetProperties> {
...
-		this._theme = event.target.checked ? dojoTheme : {};
+	themeChange(event: Event) {
+		const target: HTMLInputElement = event.target as any;
+		this._theme = target.checked ? dojoTheme : {};
```
Comment (pottedmeat) [inline src/accordionpane/example/index.ts]: This looks functionally equivalent to the button index change, but uses slightly different code.

File: src/textinput/example/index.ts [hunk 3]
```diff
@@ -39,8 +39,9 @@ export class App extends WidgetBase<WidgetProperties> {
 					type: 'text',
 					placeholder: 'Hello, World',
 					value: this._value1,
-					onChange: (event: TypedTargetEvent<HTMLInputElement>) => {
-						this._value1 = event.target.value;
+					onChange: (event: Event) => {
+						const target: HTMLInputElement = event.target as any;
```
Comment (pottedmeat) [inline src/textinput/example/index.ts]: This differs from the `as HTMLInputElement` we seem to be preferring elsewhere in the code

File: tsconfig.json [hunk 4]
```diff
@@ -14,8 +14,6 @@
 		],
 		"module": "umd",
 		"moduleResolution": "node",
-		"noImplicitAny": true,
```
Comment (pottedmeat) [inline tsconfig.json]: Has this been replaced with a different compiler option somewhere?

File: src/accordionpane/example/index.ts (hunk 1 above)
Comment (pottedmeat) [reply src/accordionpane/example/index.ts]: I mean I understand it's the function covariance stuff but I'm curious if this is our only option.

File: package.json [hunk 5]
```diff
@@ -21,16 +21,16 @@
     "test": "grunt test"
   },
   "peerDependencies": {
-    "@dojo/core": "~0.2.1",
-    "@dojo/has": "~0.1.0",
-    "@dojo/i18n": "~0.2.0",
-    "@dojo/shim": "~0.2.2",
-    "@dojo/widget-core": "~0.3.0"
+    "@dojo/core": "./tarballs/dojo-core-0.3.0-beta.1.tgz",
```
[the remark this answers is not in the document]
Comment (bryanforbes) [reply package.json]: No. This is only to update the widgets to use the unpublished API changes to the dependencies.

File: src/accordionpane/example/index.ts (hunk 1 above)
Comment (bryanforbes) [reply src/accordionpane/example/index.ts]: At this time, yes, it's our only option. @agubler and @matt-gadd talked about changing to non-bubbling events in `widget-core/vdom` (which would mean we could rely on the type of the target), so I wasn't willing to go in and update those signatures to get this type of thing to work.

File: src/accordionpane/example/index.ts (hunk 2 above)
Comment (bryanforbes) [reply src/accordionpane/example/index.ts]: Yeah, I wasn't consistent and should have been. I can clean that up next week.

File: src/textinput/example/index.ts (hunk 3 above)
Comment (bryanforbes) [reply src/textinput/example/index.ts]: See above.

File: tsconfig.json (hunk 4 above)
Comment (bryanforbes) [reply tsconfig.json]: Yes, `strict` covers `noImplicitAny`, `noImplicitThis`, `alwaysStrict`, `strictNullChecks`, and `strictFunctionTypes`.

**Comment by bitpshr** [comment]
A few minor conflicts. Thanks for doing this work.

**Comment by agubler** [comment]
I think this needs to wait until the dependencies have had their TS2.6 upgrade merged and been released.

File: src/textinput/example/index.ts [hunk 6]
```diff
@@ -39,8 +39,9 @@ export class App extends WidgetBase<WidgetProperties> {
 					type: 'text',
 					placeholder: 'Hello, World',
 					value: this._value1,
-					onChange: (event: TypedTargetEvent<HTMLInputElement>) => {
-						this._value1 = event.target.value;
+					onChange: (event: Event) => {
+						const target = event.target as HTMLInputElement;
+						this._value1 = target.value;
```
Comment (smhigley) [inline src/textinput/example/index.ts]: any reason to define `target` here instead of the `this._value1 = (event.target as HTMLInputElement).value` approach used elsewhere

Comment (bryanforbes) [reply src/textinput/example/index.ts]: I'll make that change
